sql/src/main/java/org/apache/druid/sql/calcite/planner/PlannerContext.java

```diff
@@ -179,6 +185,12 @@ public DateTimeZone getTimeZone()
     return localNow.getZone();
   }
 
+  @Nullable
+  public String getSchemaResourceType(String schema)
```

Comment:
think there would be a use case to have multiple resource types for a given schema?

Reply:
Hmm, I can't really think of a use case off the top of my head, but I guess it would be easy enough to support it. The change would go here https://github.com/apache/druid/pull/11692/files#diff-efcf39975403e19080cc5cd35af9a573608177c5ead5420ba3390c2268de0da4R38 to make it return a set instead, and then Map<String, Set<String>> through to https://github.com/apache/druid/pull/11692/files#diff-630eb92856c1b8dd3034980b486124e31952c8eef0a9195bbacb9bbe310a3023R71 which would just add all of the items in the set. https://github.com/apache/druid/pull/11692/files#diff-902bbc6a46aa1001ed51739a48b9847682c9450d7c122ce2d917370c98d4e3f9R482 would also need updated.

Reply:
if no use case, then no worries, I couldnt think of a case either. Think ok for now, up to you if you want to add support for that here now.

Reply:
This seems to return the resourceType of the resources in the schema, rather than the resourceType of the schema itself. This behavior is currently correct because all resources should be the same type in the same schema. However, I can imagine that the resources of different types can be in the same schema in the future as what other databases do. In that case, we will want to check the resourceType per resource. How about adding something like `getResourceType(String schema, String resource)` instead?

Reply:
i like this, changed :+1: